src/pkg/cli/client/byoc/baseclient.go

```diff
@@ -165,3 +166,51 @@ func (b *ByocBaseClient) GetProjectDomain(projectName, zone string) string {
 func GetPrivateDomain(projectName string) string {
 	return DnsSafeLabel(projectName) + ".internal"
 }
+
+type Obj interface {
+	Name() string
+	Size() int64
+}
+
+func (b *ByocBaseClient) ParsePulumiStackObject(ctx context.Context, obj Obj, bucket, prefix string, objLoader func(ctx context.Context, bucket, object string) ([]byte, error)) (string, error) {
+	// The JSON file for an empty stack is ~600 bytes; we add a margin of 100 bytes to account for the length of the stack/project names
+	if !strings.HasSuffix(obj.Name(), ".json") || obj.Size() < 700 {
```

Comment:
We used to return early if `obj.Name()` was `nil`, but don't now. Is that not possible anymore?

Reply:
Good catch, updated awsObj side, will no tcall this function of obj.Name or obj.Size is nill.